plugins/search/src/components/SearchModal/SearchModal.tsx

```diff
@@ -199,8 +210,15 @@ export const SearchModal = (props: SearchModalProps) => {
     >
       {open && (
         <SearchContextProvider inheritParentContextIfAvailable>
-          {(children && children({ toggleModal })) ?? (
-            <Modal toggleModal={toggleModal} />
+          {(children &&
+            children({
+              toggleModal,
+              searchResultChildren: searchResultChildren || [],
+            })) ?? (
+            <Modal
+              toggleModal={toggleModal}
+              searchResultChildren={searchResultChildren || []}
```

Comment:
The `SearchResult` component will use the default result item component if the `resultItems` is undefined:
```suggestion
              resultItems={resultItems}
```